### PR TITLE
[T4A7][W11-B4]Tay Keming Justin

### PR DESCRIPTION
**Pull request by INCENDE**

### CODE REVIEW

**Comment by INCENDE** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Name.java [hunk 1]
```diff
@@ -60,5 +61,38 @@ public boolean equals(Object other) {
...
+     * Checks the similarity of other name
+     * <br> - Case insensitive
+     * @param other
+     * @return true is similar
+     */
+    public boolean isSimilar(Name other) {
+	return isNull(other) && isEqualsIgnoreCase(other) && doesNameContainSomeParts(other);
+    }
+    
+    private boolean isNull(Name other)
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: the brace should be after function name...

File: src/seedu/addressbook/data/person/Name.java [hunk 2]
```diff
@@ -60,5 +61,38 @@ public boolean equals(Object other) {
...
+    
+    private boolean isEqualsIgnoreCase(Name other) {
+	return this.fullName.equalsIgnoreCase(other.fullName);
+    }
+    
+    private boolean doesNameContainSomeParts(Name other) {
+	String[] splitThis = this.fullName.split(" ");
+	String[] splitOther = other.fullName.split(" ");
+	
+	for(String sThis : splitThis) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: can you think of making this more efficient? :-D

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class NameTest {
+
+    /**
+     * Test the most obvious case
+     */
+    @Test 
+    public void isSimilar_totallySimilar() throws IllegalValueException {
+	Name name1 = new Name("John Smith");
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/person/NameTest.java]: going further you can consider all the steps necessary for the test to run (or preconditions) to a set up method before beginning the tests; correspondingly can put all the tear down in a method after the tests.

**Comment by okkhoy** [comment]
@INCENDE 
Looking at your commits, I can see that the order of test-develop is little messed up. I hope the tutorial discussions give an idea on how to start off with TDD.

Some comments added. Please ack & close the PR after reading comments.

**Pull request closed without merging**